### PR TITLE
Increase API version to 1.9.4 and add related changes

### PR DESCRIPTION
**Pull request by AlwinEsch**
Here the API changes for PVR to all addons for undelete recordings.

Currently only VNSI support this. On VDR stays deleted recordings still 5 Minutes in storage to allow this.

This is related to KODI Pull request https://github.com/xbmc/xbmc/pull/6367.

### CODE REVIEW

File: addons/pvr.vdr.vnsi/src/VNSIData.cpp [hunk 1]
```diff
@@ -809,6 +808,164 @@ PVR_ERROR cVNSIData::GetRecordingEdl(const PVR_RECORDING& recinfo, PVR_EDL_ENTRY
   return PVR_ERROR_NO_ERROR;
 }
 
+int cVNSIData::GetDeletedRecordingsCount()
```
Comment (FernetMenta) [inline addons/pvr.vdr.vnsi/src/VNSIData.cpp]: 1) please put the implementation into a separate commit. 
2) you need to bump vnsi protocol when adding new functions. then check if those functions are available on the backend before using those.

Comment (scarecrow420) [inline addons/pvr.vdr.vnsi/src/VNSIData.cpp]: pvr.wmc needs version bumped in its clientversion.h file as well please

**Comment by AlwinEsch** [comment]
Hello @scarecrow420, is "0.4.004" OK for there? How is a version change on WMC handled? Is it not related to connection?

If not relevant I increase it.

**Comment by scarecrow420** [comment]
dont you need to change the version on all addons that you changed?  Even if the change is only to support the new PVR API version and there are no functional changes, i believe the normal way of doing things is to bump every addon version.  I was just saying in the case of pvr.wmc whenever the addon version in addon.xml.in is bumped, it also needs to be set in clientversion.h

**Comment by AlwinEsch** [comment]
Due to a API change with two new functions and two changed ones is it not backword compatible to KODI for all addons.

For this is on all addons "<import addon="xbmc.pvr" version="1.9.4"/>" updated and this new functions added to support, but no rework on the other parts.

**Comment by AlwinEsch** [comment]
No, I also rechecked there, becomes asked and if ADDON_STATUS_LOST_CONNECTION is returned from Init the GetAddonCapabilities becomes not called.

After the backend is present and addon return no error it becomes callled. 

Only one was that it call only one time, if the backend disconnect and reconnect it become not called again.

I add there now a Version check to look for VNSI_PROTOCOLVERSION 8. What to you think, leave VNSI_RECORDINGS_DELETED_ACCESS_SUPPORTED there or remove it then?

**Comment by AlwinEsch** [comment]
Hello @opdenkamp, how are currently updates on addons handled? Are still updates possible or must be wait until binary build system is finished? Have three changes, 2 on PVR API and one on gui lib.

**Comment by opdenkamp** [comment]
hmm well we don't want to introduce last minute issues with changes in the build system pr, as that's the most important PR for PVR right now, but if that one's going to take a while to be merged still, then we can't keep this branch frozen.

**Comment by AlwinEsch** [comment]
No problem, if OK I add the other changes to here and becomes renamed to "Update PVR API to Version 1.9.4" and leave it open. If it take to long it can by merged otherwise closed.

How are the requests in KODI handled, from my side must come both together or becomes it most times performed independent?

**Comment by AlwinEsch** [comment]
Is updated.

**Comment by FernetMenta** [comment]
I talked to Montellese who said we should go on here until pvr binary PR is finished. @opdenkamp thumbs up for this?

**Comment by opdenkamp** [comment]
Sure, going out for carnaval again now ;)


-------- Oorspronkelijk bericht --------
Van: Rainer Hochecker <notifications@github.com> 
Datum:15-02-2015  12:35  (GMT+01:00) 
Aan: opdenkamp/xbmc-pvr-addons <xbmc-pvr-addons@noreply.github.com> 
Cc: Lars Op den Kamp <opdenkamp@gmail.com> 
Onderwerp: Re: [xbmc-pvr-addons] Increase API version to 1.9.4 and add related
  changes (#427) 

I talked to Montellese who said we should go on here until pvr binary PR is finished. @opdenkamp thumbs up for this?

**Comment by FernetMenta** [comment]
@AlwinEsch we need to bump addon versions. I suggest that we increase minor so backports to helix can increment micro.
please also update changelog for the addons which have this file.

**Comment by AlwinEsch** [comment]
Is updated here and on https://github.com/xbmc/xbmc/pull/6367

**Comment by AlwinEsch** [comment]
Is updated again, has seen on other places there that also partly the smallest number was increased, Sorry.

**Comment by janbar** [comment]
@AlwinEsch , please could you make the PR from your branch of pvr.mythtv ?

**Comment by janbar** [comment]
@AlwinEsch , is vdr handle recording directories ?
Trying to enable the feature, deleted recording screen shows all directories and sub-directories, containing not deleted, empty. And vis-versa: not deleted recording screen shows empty directories containing only deleted. We have to hide empty directory. Have any idea how to proper fix the issue ? I could make a PR to Kodi to fix it.

**Comment by AlwinEsch** [comment]
Has not seen this failure here sorry, was only  tested with pvr demo with folders.If you could fix it thanks, my computer system make currently problems and I must fix this :-(

**Comment by janbar** [comment]
@AlwinEsch , i would try to fix it this next week. I haven't try pvr.demo, but just enabling feature in pvr.mythtv which handles sub-directories. I let you know.
